src/kernel/src/pager.rs

```diff
@@ -0,0 +1,75 @@
+use twizzler_abi::{
+    object::ObjID,
+    pager::{CompletionToKernel, RequestFromKernel},
+};
+
+use crate::{
+    obj::{lookup_object, LookupFlags},
+    queue::{ManagedQueueReceiver, ManagedQueueSender, QueueObject},
+    sched::schedule,
+    thread::{start_new_kernel, Priority},
+};
+
+struct PagerQueues {
+    sender: Option<ManagedQueueSender<RequestFromKernel, CompletionToKernel>>,
+    receiver: Option<ManagedQueueReceiver<(), ()>>,
+}
+
+static mut PAGER_QUEUES: PagerQueues = PagerQueues {
+    sender: None,
+    receiver: None,
+};
+
+extern "C" fn pager_entry() {
```

Comment:
We still call C?

Reply:
Entry points for threads still need to be the C abi, since we set up execution for a new thread for the C abi (since it's stable, Rust ABI is not)

Reply:
OK, makes sense, thanks!